op/model.py

```diff
@@ -312,4 +339,13 @@ def config_get(self):
         return self._run('config-get')
 
     def is_leader(self):
-        return self._run('is-leader')
+        """Obtain the current leadership status for the unit the charm code is executing on.
+
+        The value is cached for the duration of a lease which is 30s in Juju.
+        """
+        if time.monotonic() - self._leader_check_time > self.LEASE_REFRESH_PERIOD or self._is_leader is None:
+            # Record the time before executing is-leader to account for the execution delay and so make the next check to happen earlier.
```

Comment:
```
# Current time MUST be saved before running is-leader to ensure the cache
# is only used inside the window that is-leader itself asserts.
```

Reply:
Will address in https://github.com/canonical/operator/pull/59